Core/Src/ws2812.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * ws2812.c
+ *
+ *  Created on: Sep 22, 2024
+ *      Author: anthony
+ */
+
+#include "stm32l4xx_hal.h"
+#include "ws2812.h"
+#include "hex_to_rgb_conversion.h"
+
+extern TIM_HandleTypeDef htim1;
+extern DMA_HandleTypeDef hdma_tim1_ch2;
+
+uint8_t led_in_cycle = 0;	// This is a value that we use to keep track of which LED is on during the cycle.
+uint8_t out_buf[BUFF_SIZE]; // We add 24 to the end so that we have 24 (one LED) bits of 0
```

Comment:
Consider whether or not you need these for your code and we might be able to get rid of led_in_cycle (maybe).

At the very least, only `uint8_t out_buf` should be at the top of the file, led_in_cycle can live in a rotate function or main.